components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +180,49 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// findDistance calculates the distance between two points on Earth.
+// lat1, lon1: Latitude and Longitude of point 1 (in decimal degrees)
+// lat2, lon2: Latitude and Longitude of point 2 (in decimal degrees)
+// returns the distance in meters.
+func findDistance(lat1, lon1, lat2, lon2 float64) float64 {
+	lat1Rad, lon1Rad := utils.DegToRad(lat1), utils.DegToRad(lon1)
+	lat2Rad, lon2Rad := utils.DegToRad(lat2), utils.DegToRad(lon2)
+
+	dLat := lat2Rad - lat1Rad
+	dLon := lon2Rad - lon1Rad
+
+	a := math.Sin(dLat/2)*math.Sin(dLat/2) +
+		math.Cos(lat1Rad)*math.Cos(lat2Rad)*
+			math.Sin(dLon/2)*math.Sin(dLon/2)
+	c := 2 * math.Atan2(math.Sqrt(a), math.Sqrt(1-a))
+
+	distanceKm := earthRadiusKm * c
+	return distanceKm * 1000 // convert km to meters
+}
+
+// calculateCompassDegreeError calculates the compass degree error
+// of two geo points.
+func (g *CachedData) calculateCompassDegreeError() float64 {
```

Comment:
I don't understand what this function does. We seem to get the distance between the last time we called `Position()` and the latest (uncached) Position, then get the angle formed by a right triangle with side lengths that and some value that depends on our fix. 

Maybe my problem is that I don't understand what the "compass degree error" is (I had assumed it was the error in the bearing, as in "we think we're facing northwest but could be off by this much"). Consider leaving a more descriptive comment on line 204; the one that's currently there says nothing useful and if it wasn't for the linter I'd ask for it to be removed.

The Jira ticket (RSDK-6731) is similarly vague about what this function is supposed to do.

Reply:
The error is th error in accuracy, not the precision from north.

This is saying:
Given that that I am pointing 67degs east, Iwith an rtk firx of 4 (a radius of 10cm around the point)  have a window of 5 degrees of error based on my last position and the circle drawn around my new position. 

This is because the RMC message that reports compass heading in a GPS rtk is based on it calculating the heading between the two points in series in time. Hope this helps.

Reply:
Yeah, that kinda helps. I better understand what this function is doing, but I have a worse understanding of why this function is useful. It now seems like the error decreases at high speeds: if your most recently-checked position is farther away than your current position (because you're moving fast), you'll have smaller errors (because the triangle whose arctangent we're calculating has a larger base and thus a smaller angle). Is that something that consumers of this function should expect? Similarly, if you stop moving, the error becomes ill-defined (which is when we return NaN). Is _that_ something users expect?

You also mentioned pointing in a certain direction, but if we're on a holonomic base, the direction we're moving and the direction we're pointing are not necessarily the same, and this error seems to involve the former and not the latter, as though your mention of which direction you're pointing is irrelevant and should be omitted. and maybe that's correct, but I can't tell (you included it, which makes me think it should be included, which makes me think it should somehow be relevant but I don't see that right now).

I think the function does what you say it's supposed to do, and maybe that's good enough. but I don't understand why anyone would ever want to call it: it doesn't provide an error on anything I'd expect to care about errors on. 

Again, I think what I'm looking for is a more detailed comment explaining some combination of what this function does, why it's useful, and when you'd want to use it.

Reply:
1) NaN is what I expect and what we return when a sensor cannot report its data. It is something I expect when I van't get data from a sensor when I have to return a numerical value. If we could return a nil value I would.
2) We're not on a base, this code is only for the GPS movement sensor with no regard to the frame transform between components. That is a bigger project and needs architecting outside exposing the errors. I was using "I" literally in my example.
3) The faster speed is actually better - the further away the readings are from each other, the more confident I am that it's not a precision error from the chip itself. We can expand our heading error calculation to include averages and time-series history. But that would be out of the scope of a simple error calculation - the point of this PR.

Reply:
Also, I think this same function is/ will be used for dual GPS, where the static difference between the two sensors in space are the inputs to the distance for the compass error calculation.

Reply:
Rand and I discussed further: something I was missing is that the GPS devices do not contain a compass, but provide compass-like data anyway using the direction of most recent travel, taken from the direction from your most recent position to your current position. This function is for getting the error on _that_ calculation! and that's why the things I thought were weird exist at all: if you're moving fast, the error on the direction you're moving is smaller, and if you stop, the error on the direction you're moving becomes ill-defined. On a holonomic base, we're still talking about the direction you're moving, so the direction you're facing isn't relevant.

Thanks to Rand for setting me straight! This needs to go in a comment in the code documenting this function.